core/src/main/java/org/apache/brooklyn/util/core/config/ConfigBag.java

```diff
@@ -126,7 +126,11 @@ public void markUsed(String key) {
     /** As {@link #newInstanceExtending(ConfigBag)} but also putting the supplied values. */
     @Beta
     public static ConfigBag newInstanceExtending(final ConfigBag configBag, Map<?,?> optionalAdditionalValues) {
-        return newInstanceExtending(configBag).putAll(optionalAdditionalValues);
```

Comment:
As per @neykov 's comments, do we really want this change?

    I think we shouldn't do this, too intrusive. Better make the merging behaviour specific to provisioning.properties.
    Or even more constrained to templateOptions.
    I like the idea of the PR, but feel that it should be more constrained in scope. Either having this behaviour only for prosivioning.properties or even more to templateOptions

It feels high risk. It's unclear what impact it will have on other users (e.g. someone writing a YAML entity that extends another entity, overriding a config key; for that particular example, we need to discuss it on dev@brooklyn a lot more first).